src/app/fdctl/run/tiles/fd_sign.c

```diff
@@ -73,35 +73,11 @@ during_frag_sensitive( void * _ctx,
   fd_sign_ctx_t * ctx = (fd_sign_ctx_t *)_ctx;
   FD_TEST( in_idx<MAX_IN );
 
-  switch( ctx->in_role[ in_idx ] ) {
-    case FD_KEYGUARD_ROLE_VOTER:
-      fd_memcpy( ctx->_data, ctx->in_data[ in_idx ], sz );
-      break;
-    case FD_KEYGUARD_ROLE_LEADER:
-      fd_memcpy( ctx->_data, ctx->in_data[ in_idx ], 32UL );
-      break;
-    case FD_KEYGUARD_ROLE_TLS:
-      fd_memcpy( ctx->_data, ctx->in_data[ in_idx ], 130UL );
-      break;
-    case FD_KEYGUARD_ROLE_GOSSIP:
-      if( sz>FD_KEYGUARD_SIGN_REQ_MTU ) {
-        FD_LOG_WARNING(("Corrupt gossip signing message with size %lu", sz));
-        *opt_filter = 1;
-        return;
-      }
-      fd_memcpy( ctx->_data, ctx->in_data[ in_idx ], sz );
-      break;
-    case FD_KEYGUARD_ROLE_REPAIR:
-      if( sz>FD_KEYGUARD_SIGN_REQ_MTU ) {
-        FD_LOG_WARNING(("Corrupt repair signing message with size %lu", sz));
-        *opt_filter = 1;
-        return;
-      }
-      fd_memcpy( ctx->_data, ctx->in_data[ in_idx ], sz );
-      break;
-    default:
-      FD_LOG_CRIT(( "unexpected link role %lu", ctx->in_role[ in_idx ] ));
+  int role = ctx->in_role[ in_idx ];
+  if( sz>FD_KEYGUARD_SIGN_REQ_MTU ) {
```

Comment:
Trivial nit but it's the sign tile so fair: this could in some weird cases copy past the end of the input buffer (if dcache size of the LEADER ink is 32UL, and attacker passes 64UL here it would pass, and copy 32 off the end);.

In practice it can't happen and wouldn't do anything anyway, but I think you should restrict this more tightly.